src/main/java/it/pagopa/pn/deliverypush/middleware/actiondao/dynamo/mapper/DtoToEntityFutureActionMapper.java

```diff
@@ -0,0 +1,53 @@
+package it.pagopa.pn.deliverypush.middleware.actiondao.dynamo.mapper;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import it.pagopa.pn.api.dto.notification.timeline.TimelineElementCategory;
+import it.pagopa.pn.deliverypush.abstractions.actionspool.Action;
+import it.pagopa.pn.deliverypush.middleware.actiondao.dynamo.FutureActionEntity;
+import it.pagopa.pn.deliverypush.middleware.actiondao.dynamo.PhysicalAddressConv;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Component
+public class DtoToEntityFutureActionMapper {
+    private final ObjectMapper objectMapper;
+    private final Map<TimelineElementCategory, ObjectWriter> objectWriters;
+
+    public DtoToEntityFutureActionMapper(ObjectMapper objectMapper, Map<TimelineElementCategory, ObjectWriter> objectWriters) {
+        this.objectMapper = objectMapper;
+        this.objectWriters = objectWriters;
+    }
+
+    public FutureActionEntity dtoToEntity(Action dto, String timeSlot) {
+        FutureActionEntity.FutureActionEntityBuilder builder = FutureActionEntity.builder()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;timeSlot&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;dto&#39;. The same parameter type is validated here for example: pn-delivery-push/src/main/java/it/pagopa/pn/deliverypush/abstractions/actionspool/impl/ActionsPoolImpl.java:123. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).